ctpolicy/chromepolicy.go

```diff
@@ -0,0 +1,130 @@
+
+// Package ctpolicy contains structs describing CT policy requirements and
+// corresponding logic.
+package ctpolicy
+
+import (
+	"github.com/google/certificate-transparency-go/loglist"
+	"github.com/google/certificate-transparency-go/x509"
+)
+
+// googGroupBuilder is used to build log-group covering Google-operated logs.
+type googGroupBuilder struct {
+}
+
+func (g googGroupBuilder) GetName() string {
+	return "Google-operated"
+}
+func (g googGroupBuilder) GetDescription() string {
+	return "All logs that are operated by Google."
+}
+func (g googGroupBuilder) IsMember(log *loglist.Log) bool {
+	return log.ByGoogle()
+}
+func (g googGroupBuilder) IsBase() bool {
+	return false
+}
+
+// nonGoogGroupBuilder is used to build log-group covering not-Google-operated logs.
+type nonGoogGroupBuilder struct {
+}
+
+func (g nonGoogGroupBuilder) GetName() string {
+	return "Non-Google-operated"
+}
+func (g nonGoogGroupBuilder) GetDescription() string {
+	return "All logs that are NOT operated by Google."
+}
+func (g nonGoogGroupBuilder) IsMember(log *loglist.Log) bool {
+	return !log.ByGoogle()
+}
+func (g nonGoogGroupBuilder) IsBase() bool {
+	return false
+}
+
+// ChromeCTPolicy incapsulates Chrome Certificate Policy. See https://github.com/chromium/ct-policy/blob/master/ct_policy.md#qualifying-certificate
+// Instantiated for a particular LogList.
+type ChromeCTPolicy struct {
+	googGroup    LogGroup
+	nonGoogGroup LogGroup
+	baseGroup    LogGroup
+	ll           *loglist.LogList
+}
+
+// Groups returns map of log-groups policy operates onto.
+func (chromeP *ChromeCTPolicy) Groups() map[string]*LogGroup {
+	m := make(map[string]*LogGroup)
+	m[chromeP.googGroup.name] = &chromeP.googGroup
+	m[chromeP.nonGoogGroup.name] = &chromeP.nonGoogGroup
+	m[chromeP.baseGroup.name] = &chromeP.baseGroup
+	return m
+}
+
+// GroupMinInclusions returns map from log-group names to minimal required number of logs within that group.
+// Numbers come from https://github.com/chromium/ct-policy/blob/master/ct_policy.md#qualifying-certificate
+func (chromeP *ChromeCTPolicy) GroupMinInclusions(cert *x509.Certificate) map[string]int {
+	minInclusions := make(map[string]int)
+	minInclusions[chromeP.googGroup.name] = 1
+	minInclusions[chromeP.nonGoogGroup.name] = 1
+	startYear, startMonth, startDay := cert.NotBefore.Date()
+	endYear, endMonth, endDay := cert.NotAfter.Date()
+	lifetimeInMonths := (int(endYear)-int(startYear))*12 + (int(endMonth) - int(startMonth))
+	if endDay < startDay {
+		// partial month
+		lifetimeInMonths--
+	}
+
+	if lifetimeInMonths < 15 {
+		minInclusions[chromeP.baseGroup.name] = 2
+	} else if lifetimeInMonths <= 27 {
+		minInclusions[chromeP.baseGroup.name] = 3
+	} else if lifetimeInMonths <= 39 {
+		minInclusions[chromeP.baseGroup.name] = 4
+	} else {
+		minInclusions[chromeP.baseGroup.name] = 5
+	}
+	return minInclusions
+}
+
+// GroupConcurrentInclusions returns map from log-group names to recommended number concurrent requests for that group.
```

Comment:
"to recommended number **of** concurrent requests"

Reply:
Added.